### PR TITLE
more API test cases for direct upload

### PR DESCRIPTION
**Pull request by individual-it**
More tests for https://community.openproject.org/projects/nextcloud-integration/work_packages/45464

### CODE REVIEW

File: tests/acceptance/features/api/directUpload.feature [hunk 1]
```diff
@@ -85,3 +85,258 @@ Feature: API endpoint for direct upload
...
+    }
+    """
+    And the content of file at "/file.txt" for user "Alice" should be "original data"
+    And the content of file at "/file (1).txt" for user "Alice" should be "new data"
+
+
+  Scenario: set overwrite to false and send file with an existing filename, also files with that name and suffixed numbers also exist
+    Given user "Alice" has uploaded file with content "data 0" to "/file.txt"
+    Given user "Alice" has uploaded file with content "data 1" to "/file (1).txt"
+    Given user "Alice" has uploaded file with content "data 2" to "/file (2).txt"
```
Comment (SwikritiT) [inline tests/acceptance/features/api/directUpload.feature:246]:
```suggestion
    And user "Alice" has uploaded file with content "data 1" to "/file (1).txt"
    And user "Alice" has uploaded file with content "data 2" to "/file (2).txt"
```

File: tests/acceptance/features/api/directUpload.feature [hunk 2]
```diff
@@ -85,3 +85,258 @@ Feature: API endpoint for direct upload
...
+    """"
+    {
+    "type": "object",
+    "required": [
+        "file_name",
+        "file_id"
+      ],
+      "properties": {
+          "file_name": {"type": "string", "pattern": "^file \(1\).txt$"},
+          "file_id": {"type" : "integer"}
```
Comment (SwikritiT) [inline tests/acceptance/features/api/directUpload.feature:235]: I noticed in nextcloud for the suffice if the filename already exists it (2) for the second file, maybe the counting starts from 1, should we also follow the same pattern?

File: tests/acceptance/features/api/directUpload.feature (hunk 1 above)
Comment (individual-it) [reply tests/acceptance/features/api/directUpload.feature]: done

File: tests/acceptance/features/api/directUpload.feature (hunk 2 above)
Comment (individual-it) [reply tests/acceptance/features/api/directUpload.feature]: done

**Review by SwikritiT — APPROVED** [review]
LGTM

**Pull request merged**